### PR TITLE
WIP: Add some OS X missing dependencies (re-render).

### LINKED ISSUE #23: gdalwarp missing libjson library...
Mac OS 10.10.5
conda 4.0.5
conda create --name py3exp python=3 -y
source activate py3exp

Then I run the following shell script:

```shell
#!/bin/sh

echo "1 2 3 4\n5 6 7 8\n9 10 11 12\n13 14 15 16" > grid.asc
echo "ncols 4\nnrows 4\nxllcenter -70.0\nyllcenter -19.0\ncellsize 5\nodata_value -9999\n" > grid.hdr
proj4="+y_0=2400761.714982585 +lat_ts=-19.6097 +a=6378137.0 +proj=merc +units=m +b=6356752.3142 +lat_0=-19.6097 +x_0=-0.0 +lon_0=-70.7691"
gdalwarp -s_srs "+proj=latlong" -t_srs "$proj4" -of EHdr grid.asc grid.flt
```

and I get the following error:

dyld: Library not loaded: /usr/local/lib/libjson-c.2.dylib
  Referenced from: /Users/mhearne/anaconda/envs/py3exp/bin/gdalwarp
  Reason: image not found
testwarp.sh: line 6: 19723 Trace/BPT trap: 5       gdalwarp -s_srs "+proj=latlong" -t_srs "$proj4" -of EHdr grid.asc grid.flt

### PR DESCRIPTION
**Pull request by ocefpaf**
Closes #23.

@mhearne-usgs I stole your example and added it as a test.

### CODE REVIEW

**Comment by ocefpaf** [comment]
Argh... I need to tell `gdal` where to find `json-c` headers. I'll do that tomorrow. Time to :zzz:

**Comment by ocefpaf** [comment]
After some digging I found out that `gdal` brings its own `json-c` and `gifib` code, but `gdal` does not use them if it finds an external source. The OS X problem must be that `homebrew` has those libraries installed. (See [this](https://travis-ci.org/conda-forge/gdal-feedstock/jobs/118709630#L626) and this [line](https://travis-ci.org/conda-forge/gdal-feedstock/jobs/118709630#L699) from the OS X logs and compare with the same [lines](https://circleci.com/gh/conda-forge/gdal-feedstock/26) from the Linux logs.)

For now I will force everything to come from conda-forge packages on Linux and OS X. I will leave Windows using the bundled code. If we find problems we can delete `homebrew` and force everything to come from the bundle code.

Pinging @pelson and @jakirkham in case you see something similar in another package.

**Comment by ocefpaf** [comment]
```shell
GDAL is now configured for x86_64-unknown-linux-gnu

  Installation directory:    /opt/conda/envs/_build
  C compiler:                gcc -g -O2 -DHAVE_SSE_AT_COMPILE_TIME
  C++ compiler:              g++ -g -O2 -DHAVE_SSE_AT_COMPILE_TIME

  LIBTOOL support:           yes

  LIBZ support:              external
  LIBLZMA support:           no
  GRASS support:             no
  CFITSIO support:           no
  PCRaster support:          internal
  LIBPNG support:            external
  DDS support:               no
  GTA support:               no
  LIBTIFF support:           external (BigTIFF=yes)
  LIBGEOTIFF support:        internal
  LIBJPEG support:           external
  12 bit JPEG:               no
  12 bit JPEG-in-TIFF:       no
  LIBGIF support:            external
  OGDI support:              no
  HDF4 support:              yes
  HDF5 support:              yes
  NetCDF support:            yes
  Kakadu support:            no
  JasPer support:            no
  OpenJPEG support:          yes
  ECW support:               no
  MrSID support:             no
  MrSID/MG4 Lidar support:   no
  MSG support:               no
  GRIB support:              yes
  EPSILON support:           no
  WebP support:              no
  cURL support (wms/wcs/...):yes
  PostgreSQL support:        yes
  MySQL support:             no
  Ingres support:            no
  Xerces-C support:          yes
  NAS support:               yes
  Expat support:             no
  libxml2 support:           no
  Google libkml support:     no
  ODBC support:              no
  PGeo support:              no
  FGDB support:              no
  MDB support:               no
  PCIDSK support:            internal
  OCI support:               no
  GEORASTER support:         no
  SDE support:               no
  Rasdaman support:          no
  DODS support:              no
  SQLite support:            no
  PCRE support:              no
  SpatiaLite support:        no
  DWGdirect support          no
  INFORMIX DataBlade support:no
  GEOS support:              yes
  Poppler support:           no
  Podofo support:            no
  OpenCL support:            no
  Armadillo support:         no
  FreeXL support:            no
  SOSI support:              no
```

File: recipe/build.sh [hunk 1]
```diff
@@ -14,9 +14,9 @@ CPPFLAGS="-I$PREFIX/include" LDFLAGS="-L$PREFIX/lib" \
             --with-xerces=$PREFIX \
             --with-netcdf=$PREFIX \
             --with-geos=$PREFIX/bin/geos-config \
-            --with-kea=$PREFIX/bin/kea-config \
```
Comment (ocefpaf) [inline recipe/build.sh]: The log says this is invalid.

**Comment by jakirkham** [comment]
Could you use Japser here?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -38,6 +38,9 @@ requirements:
         - postgresql  # [linux]
         - openjpeg  # [not win]
         - curl  # [not win]
+        - giflib  # [not win]
```
Comment (ocefpaf) [inline recipe/meta.yaml]: It is safe to remove this from `fiona` now. It belongs here.

**Comment by ocefpaf** [comment]
> Could you use JasPer here?

Testing that.

**Comment by jakirkham** [comment]
Just noticed it the other day. OpenCV likes it if we provide it so I went searching through our now over 200 package list and sure enough it exists. :smile:

**Comment by ocefpaf** [comment]
> sure enough it exists. :smile:

I probably added it :wink:

**Comment by jakirkham** [comment]
I believe it.

File: ci_support/run_docker_build.sh [hunk 3]
```diff
@@ -38,70 +40,70 @@ echo "$config" > ~/.condarc
...
     export CONDA_PY=27
     set +x
     conda build /recipe_root --quiet || exit 1
-    
     /feedstock_root/ci_support/upload_or_check_non_existence.py /recipe_root conda-forge --channel=main || exit 1
-    
-    
+
+    set -x
+    export CONDA_NPY=111
```
Comment (pelson) [inline ci_support/run_docker_build.sh]: Wooo. Swish. :+1:

**Comment by ocefpaf** [comment]
Note to self: we should definetly use

```shell
if [[ $(uname) == Darwin ]]; then
  brew remove --force $(brew list)
fi
```

when building `gdal`. Looking at the logs I noticed that `Travis-CI` OS X image has `gdal` and all its dependencies installed via homebrew. That is a major interference factor that is trowing us off in `gdal`, `fiona`, and etc.

**Comment by pelson** [comment]
I support that. Let's get it into the smithy.

**Comment by ocefpaf** [comment]
> I support that. Let's get it into the smithy.

Done in https://github.com/conda-forge/conda-smithy/pull/114

PS: Maybe we can wait until I sort out this PR and https://github.com/conda-forge/staged-recipes/pull/257 first.

File: ci_support/run_docker_build.sh [hunk 4]
```diff
@@ -38,70 +40,70 @@ echo "$config" > ~/.condarc
 conda clean --lock
 
 conda update --yes --all
-
+conda install --yes conda-build==1.18.2
```
Comment (jakirkham) [inline ci_support/run_docker_build.sh]: This wasn't already there?

Comment (ocefpaf) [inline ci_support/run_docker_build.sh]: You are thinking about the `gdal2` branch.

**Comment by ocefpaf** [comment]
Looks good. I am merging this and hopefully we areno longer contaminated by `homebrew`.

**Comment by ocefpaf** [comment]
Below is the list that was uninstalled from homebrew:

```
- apple-gcc42
- autoconf
- automake
- boost
- cgal
- cloog
- cloog-ppl015
- cloog018
- cmake
- freexl
- gcc
- gcc46
- gcc48
- gdal
- geos
- giflib
- gmp
- gmp4
- go
- gpp
- isl
- isl011
- jpeg
- json-c
- libgeotiff
- libgpg-error
- libksba
- liblwgeom
- libmpc
- libmpc08
- libpng
- libspatialite
- libtiff
- libtool
- libxml2
- libyaml
- lzlib
- maven
- mercurial
- mpfr
- mpfr2
- node
- openssl
- ossp-uuid
- pkg-config
- postgis
- postgresql
- ppl011
- proj
- pyenv
- readline
- sfcgal
- sqlite
- subversion
- wget
- xctool
```

**Comment by jakirkham** [comment]
Thanks for the list, @ocefpaf. This is great. :smile:

I'm going to add it to the PR so we can keep track of this.

**Comment by ocefpaf** [comment]
Ouch! I meant to put it there. I guess it is time :sleeping:

Thanks @jakirkham!

**Comment by jakirkham** [comment]
No worries. I ended up dropping it in the issue instead so the PR didn't get super noisy. Sounds good.

**Pull request merged**